internal/authz/mock/simple_authz.go

```diff
@@ -58,13 +59,23 @@ func (n *SimpleClient) Delete(_ context.Context, _ string, _ authz.Role, project
 }
 
 // DeleteUser implements authz.Client
-func (_ *SimpleClient) DeleteUser(_ context.Context, _ string) error {
+func (n *SimpleClient) DeleteUser(_ context.Context, _ string) error {
+	n.Assignments = nil
+	n.Allowed = nil
 	return nil
 }
 
 // AssignmentsToProject implements authz.Client
-func (_ *SimpleClient) AssignmentsToProject(_ context.Context, _ uuid.UUID) ([]*minderv1.RoleAssignment, error) {
-	return []*minderv1.RoleAssignment{}, nil
+func (n *SimpleClient) AssignmentsToProject(_ context.Context, p uuid.UUID) ([]*minderv1.RoleAssignment, error) {
+	if n.Assignments == nil {
+		return nil, nil
+	}
+
+	if _, ok := n.Assignments[p]; !ok {
+		return nil, nil
+	}
+
+	return n.Assignments[p], nil
```

Comment:
Since the default value for `[]*minderv1.RoleAssignment` is `nil`, this could just be:

```suggestion
	return n.Assignments[p], nil
```